keyboards/keychron/q11/ansi_encoder/ansi_encoder.c

```diff
@@ -0,0 +1,236 @@
+        {0,27}, {21,27}, {38,27}, {52,27}, {66,27}, {79,27}, {93,27},
+        {0,40}, {22,40}, {41,40}, {55,40}, {69,40}, {83,40}, {97,40},
+        {0,52}, {26,52},          {48,52}, {61,52}, {76,52}, {90,52}, {102,52},
+        {0,64}, {19,64}, {36,64}, {53,64}, {70,64},          {94,64},
+                  {110,0},  {124,0},  {138,0},  {152,0},  {165,0},  {179,0},  {193,0},  {207,0},
+                  {110,15}, {124,15}, {138,15}, {152,15}, {165,15}, {179,15}, {200,15},           {224,15},
+        {104,27}, {117,27}, {131,27}, {145,27}, {158,27}, {172,27}, {186,27}, {203,27},           {224,27},
+        {107,40}, {121,40}, {134,40}, {148,40}, {162,40}, {176,40},           {198,40},           {224,40},
+                  {114,52}, {127,52}, {141,52}, {155,52}, {169,52},           {188,52}, {210,52},
+                            {126,64},           {152,64}, {166,64}, {180,64}, {196,64}, {210,64}, {224,64},
+    },
+    {
+        // RGB LED Index to Flag
+        1, 1, 1, 1, 1, 1, 1,
+        1, 1, 4, 4, 4, 4, 4, 4,
+        1, 1, 4, 4, 4, 4, 4,
+        1, 8, 4, 4, 4, 4, 4,
+        1, 1,    4, 4, 4, 4, 4,
+        1, 1, 1, 1, 1,    4,
+           1, 1, 1, 1, 1, 1, 1, 1,
+           4, 4, 4, 4, 4, 4, 1,    1,
+        4, 4, 4, 4, 4, 4, 4, 4,    1,
+        4, 4, 4, 4, 4, 4,    1,    1,
+           4, 4, 4, 4, 4,    1, 1,
+              4,    1, 1, 1, 1, 1, 1,
+    }
+};
+
+// clang-format on
+
+#endif
+
+#define ADC_BUFFER_DEPTH 1
+#define ADC_NUM_CHANNELS 1
+#define ADC_SAMPLING_RATE ADC_SMPR_SMP_12P5
+#define ADC_RESOLUTION ADC_CFGR_RES_10BITS
+
+static int16_t analogReadPin_my(pin_t pin) {
+    ADCConfig adcCfg = {};
+    adcsample_t sampleBuffer[ADC_NUM_CHANNELS*ADC_BUFFER_DEPTH];
+    ADCDriver* targetDriver = &ADCD1;
+    ADCConversionGroup adcConversionGroup = {
+        .circular = FALSE,
+        .num_channels = (uint16_t)(ADC_NUM_CHANNELS),
+        .cfgr = ADC_RESOLUTION,
+    };
+
+    palSetLineMode(pin, PAL_MODE_INPUT_ANALOG);
+    switch (pin) {
+        case B0:
+            adcConversionGroup.smpr[2] = ADC_SMPR2_SMP_AN15(ADC_SAMPLING_RATE);
+            adcConversionGroup.sqr[0] = ADC_SQR1_SQ1_N(ADC_CHANNEL_IN15);
+            sampleBuffer[0] = 0;
+            break;
+        case B1:
+            adcConversionGroup.smpr[2] = ADC_SMPR2_SMP_AN16(ADC_SAMPLING_RATE);
+            adcConversionGroup.sqr[0] = ADC_SQR1_SQ1_N(ADC_CHANNEL_IN16);
+            sampleBuffer[0] = 0;
+            break;
+        default:
+             return 0;
+    }
+    adcStart(targetDriver, &adcCfg);
+    if (adcConvert(targetDriver, &adcConversionGroup, &sampleBuffer[0], ADC_BUFFER_DEPTH) != MSG_OK) {
+        return 0;
+    }
+
+    return *sampleBuffer;
+}
+
+void keyboard_post_init_kb(void) {
+    if (is_keyboard_left()) {
+        setPinOutput(A0);
+        writePinHigh(A0);
+    } else {
+        if ((analogReadPin_my(B0) > 1000) || (analogReadPin_my(B1) > 1000)) {
```

Comment:
Correct me if I'm wrong, this is to detect the presence of USB connectivity, and disable if not present?

Reply:
Yeah, I think that should be fine. But it would be great to add a note here that this is what it is doing. That way, somebody doesn't try to clean it up later, and remove this.  (something I've been guilty of, actually).

Reply:
Well, thanks!